MekHQ/src/mekhq/campaign/CurrentLocation.java

```diff
@@ -133,6 +133,13 @@ public void setJumpPath(JumpPath path) {
         jumpPath = path;
     }
 
+    public void setRecharged(Campaign campaign) {
+        DateTime currentDate = Utilities.getDateTimeDay(campaign.getCalendar());
+        double neededRechargeTime = currentPlanet.getRechargeTime(currentDate);
+        rechargeTime = neededRechargeTime;
+        campaign.addReport("Jumpship drives fully charged");
```

Comment:
Prefer moving this reporting logic out of this and into the action listener.

Also please add something like `(GM)` to the message.

Reply:
I think I've done what you wanted in e92de1337856ded47da3bf461191e11ddfd1c6be
Also I updated some of the existing report's charging typos